packages/client-core/src/hooks/useDataRepository.ts

```diff
@@ -0,0 +1,52 @@
+import { List, Table, Utf8 } from 'apache-arrow'
+import { useContext, useEffect, useState } from 'react'
+import { deserialize } from '../common/codec'
+import { BackEndContext, handlerAdapter, Target } from '../common/context'
+import { DataRepositoryItemsFilter, TableStats, Messages, DataType } from '../common/types'
+import { getDataRepositoryItemsFilterHash as getHash } from '../common/utils/hash'
+
+export type DataRepositoryItemStructure = {
```

Comment:
This is the current structure of the data repository items table. It will likely change a bit once it's ready in kiara: the types will be defined better, extra metadata columns may be provided but the main items will stay the same: `id` is the unique id of the repository data item that we can always use to refer back to this data. `alias` is the "name" of the data, provided by the user during onboarding. `type` is either `table` (tabular data) or `string` (text). Column names and types are not null only for type `table`.  Type of columns are not defined yet but will likely be mapped to  [Arrow data types](https://arrow.apache.org/docs/python/api/datatypes.html)